From: Jeremy Linton <jeremy.linton@arm.com>
Date: Wed, 26 Jul 2023 15:54:20 -0500
Subject: [PATCH] lscpu: Even more Arm part numbers (early 2023)

There have been further MIDR/part numbers published
on https://developer.arm.com, they include:

Cortex-X4, Cortex-A520, Cortex-A720, Cortex-M85,
Cortex-M55 and Cortex-R52+.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

Upstream commit 6112ade968cbe8728ca25fccdafdb1f9599424db
---
 sys-utils/lscpu-arm.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sys-utils/lscpu-arm.c b/sys-utils/lscpu-arm.c
index 0ee10d2..41cf540 100644
--- a/sys-utils/lscpu-arm.c
+++ b/sys-utils/lscpu-arm.c
@@ -79,8 +79,11 @@ static const struct id_part arm_part[] = {
     { 0xd0e, "Cortex-A76AE" },
     { 0xd13, "Cortex-R52" },
     { 0xd15, "Cortex-R82" },
+    { 0xd16, "Cortex-R52+" },
     { 0xd20, "Cortex-M23" },
     { 0xd21, "Cortex-M33" },
+    { 0xd22, "Cortex-M55" },
+    { 0xd23, "Cortex-M85" },
     { 0xd40, "Neoverse-V1" },
     { 0xd41, "Cortex-A78" },
     { 0xd42, "Cortex-A78AE" },
@@ -96,6 +99,9 @@ static const struct id_part arm_part[] = {
     { 0xd4d, "Cortex-A715" },
     { 0xd4e, "Cortex-X3" },
     { 0xd4f, "Neoverse-V2" },
+    { 0xd80, "Cortex-A520" },
+    { 0xd81, "Cortex-A720" },
+    { 0xd82, "Cortex-X4" },
     { -1, "unknown" },
 };
 
